src/lib/features/instance-stats/getEdgeInstances.e2e.test.ts

```diff
@@ -56,77 +117,122 @@ afterAll(async () => {
+    const expectedLast = await expectedForWindow(
+        db.rawDatabase,
+        last_start,
+        last_end,
+    );
+    const res = await getEdgeInstances();
+    expect(res).toMatchObject({ lastMonth: expectedLast, monthBeforeLast: 0 });
 });
 
-test('counts only month before last', async () => {
-    const { monthBeforeLastStart } = monthWindows();
-    const mid = atMidMonth(monthBeforeLastStart);
-    const late = atLateMonth(monthBeforeLastStart);
-    await db
-        .rawDatabase(TABLE)
-        .insert([...rowsForBucket(2, mid), ...rowsForBucket(5, late)]);
-    await expect(getEdgeInstances()).resolves.toEqual({
-        lastMonth: 0,
-        monthBeforeLast: 4,
-    });
+test('counts only month before last with uniformly spread load', async () => {
+    const { prev_start, prev_end } = await getBounds(db.rawDatabase);
+    await insertSpreadAcrossMonth(db.rawDatabase, prev_start, prev_end, 2, 5);
+    const expectedPrev = await expectedForWindow(
+        db.rawDatabase,
+        prev_start,
+        prev_end,
+    );
+    const res = await getEdgeInstances();
+    expect(res).toMatchObject({ lastMonth: 0, monthBeforeLast: expectedPrev });
 });
 
-test('separates months correctly when both have data', async () => {
-    const { monthBeforeLastStart, lastMonthStart } = monthWindows();
-    const pMid = atMidMonth(monthBeforeLastStart);
-    const pLate = atLateMonth(monthBeforeLastStart);
-    const lMid = atMidMonth(lastMonthStart);
-    const lLate = atLateMonth(lastMonthStart);
-
-    await db
-        .rawDatabase(TABLE)
-        .insert([
-            ...rowsForBucket(4, pMid),
-            ...rowsForBucket(6, pLate),
-            ...rowsForBucket(3, lMid),
-            ...rowsForBucket(7, lLate),
-        ]);
-
-    await expect(getEdgeInstances()).resolves.toEqual({
-        lastMonth: 5,
-        monthBeforeLast: 5,
+test('separates months correctly with different distributed loads', async () => {
+    const { prev_start, prev_end, last_start, last_end } = await getBounds(
+        db.rawDatabase,
+    );
+    await insertSpreadAcrossMonth(db.rawDatabase, prev_start, prev_end, 4, 9);
+    await insertSpreadAcrossMonth(db.rawDatabase, last_start, last_end, 3, 4);
+    const expectedPrev = await expectedForWindow(
+        db.rawDatabase,
+        prev_start,
+        prev_end,
+    );
+    const expectedLast = await expectedForWindow(
+        db.rawDatabase,
+        last_start,
+        last_end,
+    );
+    await expect(getEdgeInstances()).resolves.toMatchObject({
+        lastMonth: expectedLast,
+        monthBeforeLast: expectedPrev,
     });
 });
 
-test('ignores current month data', async () => {
-    const { thisMonthStart, lastMonthStart } = monthWindows();
-    const lMid = atMidMonth(lastMonthStart);
-    const lLate = atLateMonth(lastMonthStart);
-    const tMid = atMidMonth(thisMonthStart);
+test('ignores current month data even when current month is heavy', async () => {
```

Comment:
Why would current month being heavy affect this?

Reply:
Just to guard against window-leak bugs I guess. I can drop this if you prefer.